fiftyone/core/annotation/validate_label_schemas.py

```
@@ -381,7 +396,22 @@ def _validate_str_list_field_label_schema(
+
+    if taxonomy is not None:
+        if component != foac.DROPDOWN:
+            raise ValueError(
+                f"'{foac.TAXONOMY}' requires a '{foac.DROPDOWN}' "
+                f"'{foac.COMPONENT}' for field '{field_name}'"
+            )
+        if values is not None:
+            raise ValueError(
+                f"'{foac.TAXONOMY}' and '{foac.VALUES}' are incompatible "
+                f"settings for field '{field_name}'"
+            )
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)